ivadomed/loader/utils.py

```diff
@@ -646,25 +647,25 @@ def create_bids_dataframe(self):
             ext_ct = ('.nii.gz', '.nii')
             suffix_ct = ('ct', 'CT')
             force_index = []
-            for root, dirs, files in os.walk(path_data):
-                for file in files:
+            for path_object in path_data.glob('**/*'):
+                if path_object.is_file():
                     # Microscopy
-                    if file == "samples.tsv" or file == "samples.json":
-                        force_index.append(file)
-                    if (file.endswith(ext_microscopy) and os.path.basename(root) == "microscopy" and
-                            (root.replace(path_data, '').startswith("sub"))):
-                        force_index.append(os.path.join(root.replace(path_data, '')))
+                    if path_object.name == "samples.tsv" or path_object.name == "samples.json":
+                        force_index.append(path_object.name)
+                    if (path_object.name.endswith(ext_microscopy) and path_object.parent.name == "microscopy" and
+                            (str(path_object.parent).replace(str(path_data) + os.sep, '').startswith("sub"))):
```

Comment:
This expression SHOULD be greatly simplified. I think all it is trying to do was to 1) get folder path, 2) remove everything before path_data (via the str replace call) including the OS separator? and check that the folder starts with "sub". I believe this can be vastly simplified with Path object through parent and stem etc. Please change that here.